File: kotlin/src/main/kotlin/MergeTwo2DArray_2570.kt

fun mergeArrays(nums1: Array<IntArray>, nums2: Array<IntArray>): Array<IntArray> {
    val result = mutableListOf<IntArray>()
	var index1 = 0
	var index2 = 0
	while (index1 < nums1.size && index2 < nums2.size) {
		val firstNum = nums1[index1]
		val secondNum = nums2[index2]

		if(firstNum[0] == secondNum[0]) {
			result.add(intArrayOf(firstNum[0], firstNum[1] + secondNum[1]))
			index1++
			index2++
		}
		else if(firstNum[0] < secondNum[0]) {
			result.add(firstNum)
			index1++
		}
		else {
			result.add(secondNum)
			index2++
		}
	}

	if (index1 < nums1.size) {
		result.addAll(nums1.slice(index1 until nums1.size))
	}
	else if (index2 < nums2.size) {
		result.addAll(nums2.slice(index2 until nums2.size))
	}

	return result.toTypedArray()
}

File: python/merge_two_2d_array_2570/merge_two_2d_array_2570.py
from typing import List


class Solution:
    def mergeArrays(self, nums1: List[List[int]], nums2: List[List[int]]) -> List[List[int]]:
        result: List[List[int]] = []
        index1: int = 0
        index2: int = 0

        while index1 < len(nums1) and index2 < len(nums2):
            first_num: List[int] = nums1[index1]
            second_num: List[int] = nums2[index2]

            if first_num[0] == second_num[0]:
                result.append([first_num[0], first_num[1] + second_num[1]])
                index1 += 1
                index2 += 1
            elif first_num[0] < second_num[0]:
                result.append(first_num)
                index1 += 1
            else:
                result.append(second_num)
                index2 += 1

        if index1 < len(nums1):
            result.extend(nums1[index1::])
        elif index2 < len(nums2):
            result.extend(nums2[index2::])

        return result
        